### ISSUE TITLE
mongo storage to reduce PeerIdInvalid errors

### ISSUE THREAD

**Issue by rking32** [UsergeTeam/Userge#442]
Currently, all peer data are stored in the memory. Hence it is cleared in every restart. So better to save them in the database and fetch them when needed. With that, we can reduce `PeerIdInvalid` errors and increase user happiness :)

https://github.com/pyrogram/pyrogram/blob/master/pyrogram/storage/storage.py

**Comment by Phyco-Ninja** [comment]
🤧 what are you gonna do about Auth key duplication error?

**Comment by rking32** [comment]
will it be hard to handle?

**Comment by Phyco-Ninja** [comment]
not really, all it will need is new authentication method, something like, it extracts auth_key, dc_id, user_id and other stuffs from given pyro session key and update that in Mongo Storage. If new authentication method is implement correctly you can move all env to a collection in database. Maybe play around with this [Mongo Storage Base](https://github.com/Phyco-Ninja/LazyAF-Utils/blob/main/pyrogram_utils/mongo_storage.py) 

`bash setup` kinda thing. 🤒

**Comment by rking32** [comment]
NIce. it's very good.

**Comment by Krishna-Singhal** [comment]
PeerIdInvalid can also be reduce by using (link to profile) mention and extract user id from mention instead of using user id directly
